uberduck_ml_dev/models/tacotron2.py

```diff
@@ -550,23 +568,32 @@ def forward(self, x, input_lengths):
 
         # pytorch tensor are not reversible, hence the conversion
         input_lengths = input_lengths.cpu().numpy()
-        x = nn.utils.rnn.pack_padded_sequence(x, input_lengths, batch_first=True)
+        x = nn.utils.rnn.pack_padded_sequence(
+            x, input_lengths, batch_first=True, enforce_sorted=False
+        )
 
         self.lstm.flatten_parameters()
         outputs, _ = self.lstm(x)
 
         outputs, _ = nn.utils.rnn.pad_packed_sequence(outputs, batch_first=True)
         return outputs
 
-    def inference(self, x):
+    def inference(self, x, input_lengths):
+        device = x.device
         for conv in self.convolutions:
-            x = F.dropout(F.relu(conv(x)), self.dropout_rate, self.training)
+            x = F.dropout(F.relu(conv(x.to(device))), 0.5, self.training)
```

Comment:
is this line necessary? `device` is defined as `x.device`, so `x.to(device)` is redundant, no?

Also, why the change to hard-code the dropout rate?

Reply:
wut, i dont remember making this change at all